meshroom/core/node.py

```diff
@@ -1465,3 +1465,78 @@ def nodeFactory(nodeDict, name=None):
             node = node.upgrade()
 
     return node
+
+def templateNodeFactory(nodeDict, name=None):
```

Comment:
A bit too much duplication.
Maybe we can just add options to the nodeFactory function.
Using a functor for the filtering condition and an option to force nodes upgrade.